docs/end-user/rollout.md

```diff
@@ -1,5 +1,417 @@
 ---
 title: Rollout
 ---
+This chapter will introduce how to use Rollout Trait to perform a rolling update on Workload.
 
-TBD
+## Background
+
+### ComponentRevision
+Each modification to a Trait will produce a kubernetes controllerRevision, the default role to generate a name of kubernetes controllerRevision is: `<Component name>-<revision number>`. You can also appoint controllerRevision name by setting `spec.components[x].externalRevision`.
+
+### Support workload type
+Rollout Trait support webservice,worker and cloneset Workload.
+
+When using webservice/worker as Workload type with Rollout Trait, Workload's name will be controllerRevision's name. And when Workload's type is cloneset-service, because of clonset support in-place update Workload's name will always be component's name.
+
+
+## How to
+
+### First Deployment
+   
+Apply the YAML below to create an Application, this Application includes a Workload of type webservice with Rollout Trait, and sets controllerRevision name to express-server-v1 by setting `spec.components[0].externalRevision` field.
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: rollout-trait-test
+spec:
+  components:
+    - name: express-server
+      type: webservice
+      externalRevision: express-server-v1
+      properties:
+        image: stefanprodan/podinfo:4.0.3
+      traits:
+        - type: rollout
+          properties:
+            targetSize: 5
+            rolloutBatches:
+              - replicas: 2
+              - replicas: 3
+EOF
+```
+This Rollout Trait has two batches with total target size of 5, the first batch has 2 replicas and second batch has 3. Only when all replicas in the first batch are ready, the second batch will start to rollout.
```

Comment:
```suggestion
This Rollout Trait has target size of 5 and two rollout batches. The first batch has 2 replicas and second batch has 3. Only after all replicas in the first batch are ready, it will start to rollout the second batch.
```